app/views/users/show.html.erb

```diff
@@ -1,10 +1,14 @@
-<h1>Here is a list of posts for User<%=@user_id%></h1>
-<%
-5.times do |n|
-    %>
-    <h2>Post <%= n+1 %><h2>
-    <%
-end
-
-%>
-<%= link_to 'Back to user', users_path %>
\ No newline at end of file
+<div class='user-posts'>
+<%= render partial: 'user', collection: @user %>
+
+    <div class='bio-container'>
+    <h2>Bio </h2>
+    <p class='bio'><%= @user[0].bio%> </p>
+    </div>
+
+<%= render partial: 'post', collection: Post.where(author: @user[0]).last(3)%>
```

Comment:
- [ ] Instead of making this query afresh in this view page, the posts on the user show page should be rendered using the `recent_post` method in the User model. Kindly refactor the `collection` accordingly.

  ```
  @user.recent_post
  ```

- [ ] **[OPTIONAL]** The `recent_post` method is incorrect. It returns 3 post regardless of whom it belongs. I strongly recommend you change to this:

  ```
  def recent_post
    posts.order(created_at: :desc).limit(3)
  end
  ```

Reply:
Thank you for your constructive feedback. `posts.order(created_at: :desc).limit(3)` gives me user_id not found under post error so I used this command.  `Post.where(author:self).order(created_at: :desc).limit(3)`

Reply:
Hi @melashu I think that is because in your User model you did not define the foreign key Rails was supposed to look up.

Rails is looking for `user_id` in the post table, whereas you have `author_id`. In your User model you should have something like so:

```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```
If you call `user.posts` Rails would find the posts using the `author_id` automatically.

Reply:
Hi @TedLivist, Thank you so much, It is really helpful feedback!